avail-cli/src/main/kotlin/com/avail/tools/fileanalyzer/configuration/CommandLineConfigurator.kt

```diff
@@ -0,0 +1,328 @@
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of the contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.avail.tools.fileanalyzer.configuration
+
+import com.avail.tools.fileanalyzer.IndexedFileAnalyzer
+import com.avail.tools.fileanalyzer.configuration.CommandLineConfigurator.OptionKey.*
+import com.avail.tools.options.*
+import com.avail.utility.configuration.ConfigurationException
+import com.avail.utility.configuration.Configurator
+import java.io.File
+import java.lang.Long.parseLong
+
+/**
+ * Provides the configuration for the [IndexedFileAnalyzer]. Specifies the
+ * options that are available as arguments to the analyzer and their effects
+ * on the output.
+ *
+ * @property helpStream
+ *   The [appendable][Appendable] to which help text should be written.
+ *
+ * @author Mark van Gulik &lt;mark@availlang.org&gt;
+ *
+ * @constructor
+ *
+ * Construct a new `CommandLineConfigurator`.
+ *
+ * @param configuration
+ *   The base [IndexedFileAnalyzerConfiguration].
+ * @param commandLineArguments
+ *   The command-line arguments.
+ * @param helpStream
+ *   The [Appendable] to which help text should be written.
+ */
+class CommandLineConfigurator constructor(
+	override val configuration: IndexedFileAnalyzerConfiguration,
+	commandLineArguments: Array<String>,
+	private val helpStream: Appendable
+) : Configurator<IndexedFileAnalyzerConfiguration>
+{
+	/** The command line arguments.  */
+	private val commandLineArguments = commandLineArguments.clone()
+
+	/** Has the [configurator][CommandLineConfigurator] been run yet? */
+	private var isConfigured = false
+
+	/**
+	 * `OptionKey` enumerates the valid configuration options.
+	 *
+	 * @author Mark van Gulik &lt;mark@availlang.org&gt;
+	 */
+	internal enum class OptionKey
```

Comment:
Fantastic options, and excellent write up of the individual options and their combinations.